scripts/speech_recognition/confidence/benchmark_asr_confidence.py

```diff
@@ -0,0 +1,426 @@
+        hp_grid = ParameterGrid(params)
+        hp_grid = list(hp_grid)
+
+        logging.info(f"==============================Running a benchmarking with grid search=========================")
+        logging.info(f"Grid search size: {len(hp_grid)}")
+        logging.info(f"Results will be written to:\nreport file `{report_file}`\nand plot directories near the file")
+        logging.info(f"==============================================================================================")
+
+        with open(report_file, "tw", encoding="utf-8") as f:
+            f.write(report_legend)
+            f.flush()
+            for i, hp in enumerate(hp_grid):
+                logging.info(f"Run # {i + 1}, grid: `{hp}`")
+                asr_model.change_decoding_strategy(apply_parameters(asr_model.cfg.decoding, hp))
+                param_list, experiment_name = get_experiment_params(asr_model.cfg.decoding.confidence_cfg)
+                plot_dir = work_dir / Path(experiment_name)
+                results = run_benchmark(
+                    asr_model,
+                    cfg.batch_size,
+                    cfg.num_workers,
+                    is_rnnt,
+                    cfg.target_level,
+                    filepaths,
+                    reference_texts,
+                    plot_dir,
+                    autocast,
+                )
+                for level, result in results.items():
+                    f.write(f"{model_typename},{','.join(param_list)},{level},{','.join([str(r) for r in result])}\n")
+                    f.flush()
+    else:
+        asr_model.change_decoding_strategy(
+            RNNTDecodingConfig(fused_batch_size=-1, strategy="greedy_batch", confidence_cfg=cfg.confidence_cfg)
+            if is_rnnt
+            else CTCDecodingConfig(confidence_cfg=cfg.confidence_cfg)
+        )
+        param_list, experiment_name = get_experiment_params(asr_model.cfg.decoding.confidence_cfg)
+        plot_dir = work_dir / Path(experiment_name)
+
+        logging.info(f"==============================Running a single benchmarking===================================")
+        logging.info(f"Results will be written to:\nreport file `{report_file}`\nand plot directory `{plot_dir}`")
+
+        with open(report_file, "tw", encoding="utf-8") as f:
+            f.write(report_legend)
+            f.flush()
+            results = run_benchmark(
+                asr_model,
+                cfg.batch_size,
+                cfg.num_workers,
+                is_rnnt,
+                cfg.target_level,
+                filepaths,
+                reference_texts,
+                plot_dir,
+                autocast,
+            )
+            for level, result in results.items():
+                f.write(f"{model_typename},{','.join(param_list)},{level},{','.join([str(r) for r in result])}\n")
+    logging.info(f"===========================================Done===============================================")
```

Comment:
This script is very good, but we need it to be part of the Jenkinstest to make sure it does not break in the future. In addition, in another PR, please add a whole battery of tests for these classes and code bases (unit tests for individual metrics and measures etc, tests for instantiating a trained model (small conformed ctc or rnnt with the "with_downloads" pytest marker) which then uses a small amount of data (already exists in unittests) to compute and run this inference) to make sure the code works properly